packages/grafana-data/src/transformations/transformers/filterByValue.ts

```diff
@@ -78,8 +78,16 @@ export const filterByValueTransformer: DataTransformerInfo<FilterByValueTransfor
               },
             };
           } else if (filter.config.id === ValueMatcherID.regex) {
-            // Due to colliding syntaxes, interpolating regex filters will cause issues.
-            return filter;
+            // Due to colliding syntaxes, interpolating regex filters may cause issues.
+            let value = filter.config.options.value;
+            if (typeof filter.config.options.value === 'string') {
+              value = ctx.interpolate(value);
+            }
```

Comment:
Could it be worth adding an extra test to protect this functionality? As we're adding an extra code-branch here 🙏

Reply:
done :)